src/main/java/uk/gov/companieshouse/api/strikeoffobjections/interceptor/ObjectionStatusInterceptor.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.companieshouse.api.strikeoffobjections.interceptor;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.servlet.HandlerInterceptor;
+import uk.gov.companieshouse.api.strikeoffobjections.common.ApiLogger;
+import uk.gov.companieshouse.api.strikeoffobjections.model.entity.Objection;
+import uk.gov.companieshouse.api.strikeoffobjections.model.entity.ObjectionStatus;
+import uk.gov.companieshouse.api.strikeoffobjections.service.impl.ERICHeaderFields;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ObjectionStatusInterceptor implements HandlerInterceptor {
+
+    private static final String OBJECTION_STATUS_INVALID = "Objection is not in a valid state for this operation. Expected a status of OPEN but was %s";
+
+    private final ApiLogger apiLogger;
+
+    public ObjectionStatusInterceptor(ApiLogger apiLogger) {
+        this.apiLogger = apiLogger;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+        final String requestId = request.getHeader(ERICHeaderFields.ERIC_REQUEST_ID);
+        apiLogger.debugContext(requestId, "Checking provided company number matches objection company number");
+        final Objection objection = (Objection) request.getAttribute(InterceptorConstants.OBJECTION_ATTRIBUTE);
+
+        // Operations on objections via the API REST interface are only allowed whilst the objection is
+        // still 'open', i.e. has not yet been submitted (to CHIPS) for processing
+        if (ObjectionStatus.OPEN != objection.getStatus()) {
+            apiLogger.errorContext(
```

Comment:
Similar to my other comment, I know this was existing behaviour, but does it need an error log? This is defined behaviour for the application, maybe an info log?